controllers/components/modelcontroller/modelcontroller_actions.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modelcontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// early exist
+	_, ok := rr.Instance.(*componentsv1.ModelController)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.ModelController)", rr.Instance)
+	}
+	rr.Manifests = []odhtypes.ManifestInfo{}
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       odhdeploy.DefaultManifestPath,
+		ContextDir: ComponentName,
+		SourcePath: "base",
+	})
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// TODO : uncomment once kserve is in
+	// mc, ok := rr.Instance.(*componentsv1.Kserve)
+	// if !ok {
+	mc, ok := rr.Instance.(*componentsv1.ModelMeshServing)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Kserve nor componetsv1.ModelMeshServing)", rr.Instance)
+	}
+	// }
+
+	if mc.Spec.DevFlags == nil {
+		return nil
+	}
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(mc.Spec.DevFlags.Manifests) != 0 {
+		for _, subcomponent := range mc.Spec.DevFlags.Manifests {
+			if strings.Contains(subcomponent.URI, ComponentName) {
+				// Download odh-model-controller
+				if err := odhdeploy.DownloadManifests(ctx, ComponentName, subcomponent); err != nil {
+					return err
+				}
+				// If overlay is defined, update paths
+				if subcomponent.SourcePath != "" {
+					rr.Manifests[0].SourcePath = subcomponent.SourcePath
+				}
+			}
+		}
+	}
+	// TODO: Implement devflags logmode logic
+	return nil
+}
+
+func updatePodSecurityRolebinding(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
```

Comment:
we have a shared action for this

Reply:
yep this is not updated since dashbaord's change, let me remove it